src/main/java/com/blog/stream/pagination/PageFetcher.java

```diff
@@ -1,7 +0,0 @@
-package com.blog.stream.pagination;
-
-import org.springframework.data.domain.Page;
-
-public interface PageFetcher<T> {
-    Page<T> fetch(final int offset, final int pageSize);
```

Comment:
Comes down to preference a little on this, whether you prefer BiConsumers over this.

My intention initially was to make this a little more understandable than BiConsumer<Int, Int, T>.
Whats your thinking behind Consumers over customer interfaces?